pkg/body-based-routing/server/runserver.go

```diff
@@ -0,0 +1,121 @@
+/*
+Copyright 2025 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package server
+
+import (
+	"context"
+	"crypto/rand"
+	"crypto/rsa"
+	"crypto/tls"
+	"crypto/x509"
+	"crypto/x509/pkix"
+	"encoding/pem"
+	"math/big"
+	"time"
+
+	extProcPb "github.com/envoyproxy/go-control-plane/envoy/service/ext_proc/v3"
+	"github.com/go-logr/logr"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+	"sigs.k8s.io/gateway-api-inference-extension/pkg/body-based-routing/handlers"
+	"sigs.k8s.io/gateway-api-inference-extension/internal/runnable"
+)
+
+// ExtProcServerRunner provides methods to manage an external process server.
+type ExtProcServerRunner struct {
+	GrpcPort int
+}
+
+// Default values for CLI flags in main
+const (
+	DefaultGrpcPort = 9002  // default for --grpcPort
+)
+
+func NewDefaultExtProcServerRunner() *ExtProcServerRunner {
+	return &ExtProcServerRunner{
+		GrpcPort: DefaultGrpcPort,
+	}
+}
+
+// AsRunnable returns a Runnable that can be used to start the ext-proc gRPC server.
+// The runnable implements LeaderElectionRunnable with leader election disabled.
+func (r *ExtProcServerRunner) AsRunnable(logger logr.Logger) manager.Runnable {
+	return runnable.NoLeaderElection(manager.RunnableFunc(func(ctx context.Context) error {
+		cert, err := createSelfSignedTLSCertificate(logger)
+		if err != nil {
+			logger.Error(err, "Failed to create self signed certificate")
+			return err
+		}
+		creds := credentials.NewTLS(&tls.Config{Certificates: []tls.Certificate{cert}})
+
+		srv := grpc.NewServer(grpc.Creds(creds))
+		extProcPb.RegisterExternalProcessorServer(
+			srv,
+			handlers.NewServer(),
+		)
+
+		// Forward to the gRPC runnable.
+		return runnable.GRPCServer("ext-proc", srv, r.GrpcPort).Start(ctx)
+	}))
+}
+
+// TODO(rramkumar1): This function is also implemented on the EPP side. We should share code.
```

Comment:
Mind creating a follow up issue for this instead of the comment?

Reply:
Done